Sources/SwiftBuild/SWBBuildParameters.swift

```diff
@@ -36,21 +36,131 @@ public struct SWBBuildParameters: Codable, Sendable {
 
 /// Refer to `SWBProtocol.RunDestinationInfo`
 public struct SWBRunDestinationInfo: Codable, Sendable {
-    public var platform: String
-    public var sdk: String
-    public var sdkVariant: String?
+    public struct SWBBuildTarget: Codable, Sendable {
+        var _internalBuildTarget: InternalBuildTarget
+
+        public static func toolchainSDK(platform: String, sdk: String, sdkVariant: String?) -> SWBBuildTarget {
+            SWBBuildTarget(_internalBuildTarget: .toolchainSDK(platform: platform, sdk: sdk, sdkVariant: sdkVariant))
+        }
+
+        public static func swiftSDK(sdkManifestPath: String, triple: String) -> SWBBuildTarget {
+            SWBBuildTarget(_internalBuildTarget: .swiftSDK(sdkManifestPath: sdkManifestPath, triple: triple))
+        }
+
+        public func asToolchainSDK() -> (platform: String, sdk: String, sdkVariant: String?)? {
+            guard case let .toolchainSDK(platform: platform, sdk: sdk, sdkVariant: sdkVariant) = _internalBuildTarget else {
+                return nil
+            }
+
+            return (platform: platform, sdk: sdk, sdkVariant: sdkVariant)
+        }
+
+        public func asSwiftSDK() -> (sdkManifestPath: String, triple: String)? {
+            guard case let .swiftSDK(sdkManifestPath: sdkManifestPath, triple: triple) = _internalBuildTarget else {
+                return nil
+            }
+
+            return (sdkManifestPath: sdkManifestPath, triple: triple)
+        }
+    }
+
+    public var buildTarget: SWBBuildTarget
     public var targetArchitecture: String
     public var supportedArchitectures: [String]
     public var disableOnlyActiveArch: Bool
     public var hostTargetedPlatform: String?
 
+    public enum CodingKeys: CodingKey {
+        case buildTarget
+        case targetArchitecture
+        case supportedArchitectures
+        case disableOnlyActiveArch
+        case hostTargetedPlatform
+
+        // These are the old coding keys that were previously associated with toolchain SDK's
+        case platform
+        case sdk
+        case sdkVariant
+    }
+
+    public init(from decoder: any Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        do {
```

Comment:
Same comments as in MessageSupport.swift re decodeIfPresent instead of do-catch, and decodeIfPresent vs decode(String?)